restful.module

```diff
@@ -561,3 +564,18 @@ function restful_date_time_format_element_validate($element, &$form_state) {
     )));
   }
 }
+
+/**
+ * Implements hook_restful_resource_alter().
+ */
+function restful_restful_resource_alter(ResourceInterface &$resource) {
```

Comment:
If we already decided to go with a hook (instead of the decorators doing their own registration themself), I think that maybe we should provide a hook - where those plugins return the class, and keep the alter as a way for implementing modules to remove/ swap some classes?